File: src/test_helpers/test_helpers.go
package test_helpers

import (
	"io/ioutil"
	"log"
	"time"
)

func CreateTmpFolder(prefix string) (newFolder string) {
	newFolder, err := ioutil.TempDir("", "dhb-"+prefix+"-")
	if err != nil {
		log.Fatal(err)
	}
	return newFolder
}

func FileContentsMatches(file1Path string, file2Path string) (bool, error) {
	file1Contents, err := readContents(file1Path)
	if err != nil {
		return false, err
	}
	file2Contents, err := readContents(file2Path)
	if err != nil {
		return false, err
	}
	return file1Contents == file2Contents, nil
}

func readContents(path string) (string, error) {
	contents, err := ioutil.ReadFile(path)
	if err != nil {
		return "", err
	}
	return string(contents), nil
}

// returns a function that always returns the same time
func TimeFixer() func() time.Time {
	fixedTime := time.Now()
	return func() time.Time {
		return fixedTime
	}
}

File: src/test_helpers/test_helpers.rs
use chrono::{DateTime, Utc};
use rand::Rng;
use std::env;
use std::fs::{self};
use std::io::{self};

pub fn create_tmp_folder(prefix: &str) -> io::Result<String> {
	let mut rng = rand::rng();
	let random_suffix: u32 = rng.random();
	let dir = env::temp_dir().join(format!("dhb-{}-{}", prefix, random_suffix));
	fs::create_dir_all(&dir)?;
	Ok(dir.to_string_lossy().into_owned())
}

// Returns a function that always returns the same time
pub fn time_fixer() -> impl Fn() -> DateTime<Utc> {
	let fixed_time = Utc::now();
	move || fixed_time
}
